diode-react/src/main/scala/diode/react/ReactConnector.scala

```diff
@@ -115,22 +116,24 @@ trait ReactConnector[M <: AnyRef] { circuit: Circuit[M] =>
       }
 
       private def changeHandler(cursor: ModelRO[S]): Unit = {
-        // modify state if we are mounted and state has actually changed
-        if (t.isMounted() && modelReader =!= t.accessDirect.state) {
-          t.accessDirect.setState(modelReader())
-        }
+        val isMounted = t.isMounted
```

Comment:
This is super awkward – I am confident that you will find a smoother way of composing these callbacks...